src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenExpression.vb

```diff
@@ -0,0 +1,519 @@
+        Return CSByte(Fix(number))
+    End Function
+    Public Shared Function SingleToByte(number as Single) As Byte
+        Return CByte(Fix(number))
+    End Function
+    Public Shared Function DoubleToByte(number as Double) As Byte
+        Return CByte(Fix(number))
+    End Function
+    Public Shared Function SingleToShort(number as Single) As Short
+        Return CShort(Fix(number))
+    End Function
+    Public Shared Function DoubleToShort(number as Double) As Short
+        Return CShort(Fix(number))
+    End Function
+    Public Shared Function SingleToUShort(number as Single) As UShort
+        Return CUShort(Fix(number))
+    End Function
+    Public Shared Function DoubleToUShort(number as Double) As UShort
+        Return CUShort(Fix(number))
+    End Function
+    Public Shared Function SingleToInteger(number as Single) As Integer
+        Return CInt(Fix(number))
+    End Function
+    Public Shared Function DoubleToInteger(number as Double) As Integer
+        Return CInt(Fix(number))
+    End Function
+    Public Shared Function SingleToUInteger(number as Single) As UInteger
+        Return CUInt(Fix(number))
+    End Function
+    Public Shared Function DoubleToUInteger(number as Double) As UInteger
+        Return CUInt(Fix(number))
+    End Function
+    Public Shared Function SingleToLong(number as Single) As Long
+        Return CLng(Fix(number))
+    End Function
+    Public Shared Function DoubleToLong(number as Double) As Long
+        Return CLng(Fix(number))
+    End Function
+    Public Shared Function SingleToULong(number as Single) As ULong
+        Return CULng(Fix(number))
+    End Function
+    Public Shared Function DoubleToULong(number as Double) As ULong
+        Return CULng(Fix(number))
+    End Function
+    Public Shared Sub Main()
+        CheckSingle(Integer.MinValue - 65F, Integer.MinValue)
+        CheckSingle(Integer.MinValue +  0F, Integer.MinValue)
+        CheckSingle(Integer.MinValue + 64F, Integer.MinValue)
+        CheckSingle(Integer.MinValue + 65F, -2147483520)
+        CheckSingle(1.99F, 1)
+        CheckSingle(Integer.MaxValue - 65F, 2147483520)
+        CheckSingle(Integer.MaxValue - 64F, Integer.MinValue)  ' overflow
+        CheckSingle(Integer.MaxValue -  0F, Integer.MinValue)  ' overflow
+
+        CheckDouble(Integer.MinValue - 1D, Integer.MinValue)  ' overflow
+        CheckDouble(Integer.MinValue + 0D, Integer.MinValue)
+        CheckDouble(Integer.MinValue + 1D, -2147483647)
+        CheckDouble(1.99D, 1)
+        CheckDouble(Integer.MaxValue - 1D, 2147483646)
+        CheckDouble(Integer.MaxValue + 0D, Integer.MaxValue)
+        CheckDouble(Integer.MaxValue + 1D, Integer.MinValue)  ' overflow
+        CheckDouble(Integer.MaxValue + 2D, Integer.MinValue)  ' overflow
+
+        Console.WriteLine("done")
+    End Sub
+
+    Public Shared Sub CheckSingle(s As Single, expected As Integer)
+        Dim result As Integer = SingleToInteger(s)
+        If result &lt;> expected
```

Comment:
>&lt;> [](start = 18, length = 5)

Consider using CDATA around the entire code snippet to avoid using XML entities and make source code for the test scenario readable